### PR TITLE
Ensure releases have a valid version to be public

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: bedrock/releasenotes/models.py [hunk 1]
```diff
@@ -211,7 +214,9 @@ def refresh(self):
                         data["channel"] = "ESR"
                     # make all releases public on non-production environments
                     if settings.DEV:
-                        data["is_public"] = True
+                        # Make sure the version is valid and publicly accessible.
```
Comment (pmac) [inline bedrock/releasenotes/models.py:217]: I'm thinking more like we should put something like this at the top of the loop:

```python
if not version_regex.match(data["version"]):
    continue
```

Comment (robhudson) [reply bedrock/releasenotes/models.py:217]: I could update it. I wasn't sure if there was possibly a reason to have those in the db for other reasons and just set them to not public. Vs not including them at all. But, smaller db the better if we don't need them.

Comment (robhudson) [reply bedrock/releasenotes/models.py]: Updated

Comment (pmac) [reply bedrock/releasenotes/models.py]: I didn't think there was a way to see it at all if the regex didn't match. Can't think of a reason to have it in the DB if it's inaccessible.

**Review by pmac — APPROVED** [review]
💯

**Pull request merged**